File: src/plugins/meson-templates/resources/src/window.py
{{include "license.py"}}

from gi.repository import Gtk


@Gtk.Template(resource_path='{{appid_path}}/{{ui_file}}')
class {{PreFix}}Window(Gtk.ApplicationWindow):
    __gtype_name__ = '{{PreFix}}Window'

    label = Gtk.Template.Child()

    def __init__(self, **kwargs):
        super().__init__(**kwargs)

File: src/plugins/meson-templates/resources/src/window.rs
use gtk::prelude::*;

pub struct Window {
    pub widget: gtk::ApplicationWindow,
}

impl Window {
    pub fn new() -> Self {
        let builder = gtk::Builder::from_resource("{{appid_path}}/{{ui_file}}");
        let widget: gtk::ApplicationWindow = builder
            .object("window")
            .expect("Failed to find the window object");

        Self { widget }
    }
}
